### PR TITLE
adapter: add database and search columns to mz_recent_activity_log

### LINKED ISSUE #25869: activity log: add session vars database and schema at time of execution
Right now the `mz_internal.mz_sql_text` catalog table uses the raw sql text that the user types it. We also don't record the database and schema set for the session at the time of query execution. As such, it's not possible to detect from the activity log cases where the user is running the query from the wrong database/schema.

One option is to record in `mz_recent_activity_log` the database and schema that were set for the session at the time of query execution.

An alternative proposed is to have a version of the sql text with the fully-qualified object names in mz_sql_text.
Note: I think it's still helpful to keep the original sql text as the official record of what was typed.

### PR DESCRIPTION
**Pull request by maddyblue**
Fixes #25869

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Add `database_name` and `search_path` to `mz_internal.mz_recent_activity_log`.

### CODE REVIEW

**Review by benesch — COMMENTED** [review]
LGTM from a SQL council perspective.

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ granted the [`mz_monitor` role](/manage/access-control/manage-roles#builtin-role
...
 | `cluster_name`             | [`text`]                     | The name of the cluster with ID `cluster_id` at execution time.                                                                                                                                                                                                               |
+| `database_name`            | [`text`]                     | The value of the session's `database` variable at execution time.                                                                                                                                                                                                               |
+| `search_path`              | [`text list`]                | The value of the session's `search_path` variable at execution time.                                                                                                                                                                                                               |
```
Comment (morsapaes) [inline doc/user/content/sql/system-catalog/mz_internal.md:61]:
```suggestion
| `database_name`            | [`text`]                     | The value of the `database` configuration parameter at execution time.                                                                                                                                                                                                               |
| `search_path`              | [`text list`]                | The value of the `search_path` configuration parameter at execution time.                                                                                                                                                                                                               |
```

Comment (morsapaes) [reply doc/user/content/sql/system-catalog/mz_internal.md:61]: For consistency with other fields.

**Review by jkosh44 — APPROVED** [review]
LGTM

**Comment by jkosh44** [comment]
Should we consider dumping all session configuration parameters into a JSON column?

**Comment by benesch** [comment]
> Should we consider dumping all session configuration parameters into a JSON column?

(If we do this, please use a `map[text => text]` type, not a JSON type! No opinions on whether it's a good idea or not.)

**Comment by maddyblue** [comment]
I don't have a strong opinion on that. I think it should be yes (use a json map) if we can identify other session vars that are required to execute the query in the same way (so that anyone viewing this in the query history can fully reproduce its behavior). We already have the cluster, database, and search_path and perhaps 3 is the magic number after which we should do something with a map. Looking through all of `show all` the only other thing that kinda jumps out is `cluster_replica`. Most other things are system settings or display preferences. I think leaving them as distinct columns is fine because it seems like we won't need more.

**Pull request merged**